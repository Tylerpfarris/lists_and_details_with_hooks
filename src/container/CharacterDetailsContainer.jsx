import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { fetchById } from '../services/avatarApi';
import Details from '../components/characters/Details';

function CharacterDetailsContainer({match}) {
  const [character, setCharacter] = useState([]);
  const [loading, setLoading] = useState(true);


  useEffect(() => {
    const id = match.params.id;
     console.log(id);

    fetchById(match.params.id)
      .then(setCharacter)
      .finally(() => setLoading(false));
  }, []);

  return loading ? (
    <h1>Loading...</h1>
  ) : (
    <>
      <h1>character</h1>
        <Details character={character}/>
    </>
  );
}

CharacterDetailsContainer.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default CharacterDetailsContainer;
