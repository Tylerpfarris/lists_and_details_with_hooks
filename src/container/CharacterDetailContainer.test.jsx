import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import CharacterDetailsContainer from './CharacterDetailsContainer'



const server = setupServer(
  rest.get(
     'https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68d1', (req, res, ctx) => {
        return res(
          ctx.json(
            {
              allies: ['Riley'],
              enemies: [],
              _id: '5cf5679a915ecad153ab68d1',
              photoUrl:
                'https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201',
              name: 'Analay',
              gender: 'Female',
              eye: 'Purple',
              hair: 'Black with red highlights',
              weapon: 'Shadows',
              affiliation: 'Unnamed team',
              first: '"',
            },
          )
        );
    }
  )
);

describe('CharacterDetailsContainer', () => {

   beforeAll(() => server.listen());
   afterAll(() => server.close());

   it('displays a characters details', async () => {
      render(
         <MemoryRouter>
            <CharacterDetailsContainer
               match={{
                  params: {
                     id: '5cf5679a915ecad153ab68d1',
                  },
               }}/>
         </MemoryRouter>
   );

      screen.getByText('Loading...')
      const detailsDiv = await screen.findByTestId('details')
      expect(detailsDiv).toMatchSnapshot()
      expect(detailsDiv).not.toBeEmptyDOMElement()

      const charName = screen.findByTestId('nameEl')
      expect(charName).toContainHTML
   })
})