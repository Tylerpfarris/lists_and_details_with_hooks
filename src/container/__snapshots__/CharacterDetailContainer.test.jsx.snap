// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetailsContainer displays a characters details 1`] = `
<div
  data-testid="details"
>
  <img
    alt="Analay"
    src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
  />
  <h3
    data-testid="nameEl"
  >
    Analay
  </h3>
  <p>
    gender: 
    Female
  </p>
  <p>
    hair color: 
    Black with red highlights
  </p>
  <p>
    weapon: 
    Shadows
  </p>
  <p>
    profession: 
    N/A
  </p>
  <p>
    affiliation: 
    Unnamed team
  </p>
  <p>
    allies: 
    Riley
  </p>
  <p>
    enemies: 
  </p>
</div>
`;
